src/reuse/modules/mobile/device.ts

```diff
@@ -14,42 +18,342 @@ export class Device {
+   * @param {string} strategy - Strategy to use for hiding the keyboard ('pressKey', 'tapOutside', 'swipeDown').
+   * @param {string} key - Key to press if using the 'pressKey' strategy (e.g., 'Done', 'Enter').
+   * @param {number} keyCode - Key code for Android (optional).
+   * @param {number} timeout - Timeout in milliseconds for retrying to hide the keyboard (default: 5000ms).
+   * @returns {Promise<void>}
+   */
+  async hideKeyboard(strategy: hideKeyboardStrategy, key?: string, keyCode?: number, timeout: number = 5000): Promise<void> {
+    const vl = this.vlf.initLog(this.hideKeyboard);
+
+    const startTime = Date.now();
+    while (Date.now() - startTime < timeout) {
+      try {
+        if (await util.browser.isAndroid()) {
+          console.log("Hiding keyboard on Android.");
+          vl.log("Hiding keyboard on Android.");
+          await browser.hideKeyboard(strategy, key, keyCode);
+        } else if (await util.browser.isIos()) {
+          console.log("Hiding keyboard on iOS.");
+          vl.log("Hiding keyboard on iOS.");
+          await browser.execute("mobile: hideKeyboard", { strategy });
+        } else {
+          console.warn("Unsupported platform: Unable to hide the keyboard.");
+          return;
+        }
+        console.log("Keyboard hidden successfully.");
+        return; // Exit if the keyboard is successfully hidden
+      } catch (error) {
+        console.warn("Failed to hide the keyboard. Retrying...", error);
+        // Wait briefly before retrying
+        await new Promise((resolve) => setTimeout(resolve, 500));
+      }
+    }
+    console.error(`Failed to hide the keyboard within the timeout of ${timeout}ms.`);
+  }
+
+  /**
+   * @function hideKeyboard
+   * @memberof mobile.device
+   * @description Checks if the keyboard is visible or not on the mobile device.
+   * @returns {Promise<boolean>}
+   */
+  async isKeyboardVisible(): Promise<boolean> {
+    const vl = this.vlf.initLog(this.isKeyboardVisible);
+
+    let isKeyboardVisible: boolean = false;
+    try {
+      if (await util.browser.isAndroid()) {
+        vl.log("check if the screen height is reduced due to the keyboard");
+        // For Android, check if the screen height is reduced due to the keyboard
+        const windowRect = await browser.getWindowRect();
+        const screenSize = await browser.getWindowSize();
+        // If the visible height is less, keyboard is visible
+        vl.log("if visible height is less, keyboard is visible");
+        isKeyboardVisible = windowRect.height < screenSize.height;
+        return isKeyboardVisible;
+      } else if (await util.browser.isIos()) {
+        // For iOS, check if the keyboard is displayed via Appium's mobile API
+        const isKeyboardShown = await browser.execute("mobile: isKeyboardShown");
+        isKeyboardVisible = isKeyboardShown === true; // Returns true if the keyboard is visible
+        return isKeyboardVisible;
```

Comment:
Maybe we can just do it like this?
```suggestion
        const isKeyboardShown = await browser.execute("mobile: isKeyboardShown");
        return isKeyboardShown;
```

Reply:
Even though ```isKeyboardShown``` is expected to return a boolean, comparing it explicitly with true adds clarity and ensures no unexpected truthy or falsy behavior (e.g., null, undefined, or other values).

Reply:
Then I would say you don't need to return `isKeyboardVisible` value from any of the if or else blocks, since you return it in the end of the isKeyboardVisible method either way. Otherwise I don't see a point of having `isKeyboardVisible` declared and assigned a default value in the beginning of the method. 

So I would go with one of the two options:

1. Remove the isKeyboardVisible variable and return the comparison results of the if and else-if statements like this:

- first if block: `return windowRect.height < screenSize.height`;
- second if block: `return isKeyboardShown === true`;

2. Remove the return statements from the if blocks, since the isKeyboardVisible will be returned anyway in the end of the method

I would prefer the second one, since it is less effort and code would be more self-explanatory (as per my subjective view)

Reply:
Yes, I agree with your point, as per your suggestions it updated the method.